runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -6937,16 +6937,57 @@ J9::Z::TreeEvaluator::VMifInstanceOfEvaluator(TR::Node * node, TR::CodeGenerator
    return NULL;
    }
 
+TR::Register *
+J9::Z::TreeEvaluator::generateCheckForValueTypeMonitorEnterOrExit(TR::Node *node, TR::Register *objReg, TR::LabelSymbol* mergeLabel, TR::LabelSymbol *helperCallLabel, TR::CodeGenerator *cg)
+   {
+   TR::Register *objectClassReg = cg->allocateRegister();
+   TR::TreeEvaluator::genLoadForObjectHeadersMasked(cg, node, objectClassReg, generateS390MemoryReference(objReg, TR::Compiler->om.offsetOfObjectVftField(), cg), NULL);
+   TR::Register *classFlagsRegister = cg->allocateRegister();
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)(cg->comp()->fe());
```

Comment:
This should be a `static_cast` if possible.

Reply:
Fixed in https://github.com/eclipse/openj9/compare/097726868780ab77a37f3f823e2ba1d82900e0db..46e59d80a6c270bd3654d431e7c3372ec3d240b8